src/json.hpp

```diff
@@ -15289,6 +15289,18 @@ inline nlohmann::json operator "" _json(const char* s, std::size_t n)
     return nlohmann::json::parse(s, s + n);
 }
 
+/*!
+@brief user-defined to_string function for JSON values
+
+This function implements a user-defined to_string  for JSON objects. 
+
+@param[in] j  a JSON object
+@return a std::string object
+*/
+inline std::string to_string(const nlohmann::json& j) {
```

Comment:
Just thought about this, but I think you should make a generic overload for every `basic_json` type (i.e. a templated overload).

Without it, users who do not use the default json type won't be able to use `to_string`.